internal/pkg/aws/s3/s3.go

```diff
@@ -78,25 +79,30 @@ func (s *S3) ZipAndUpload(bucket, key string, files ...NamedBinary) (string, err
 	if err := w.Close(); err != nil {
 		return "", err
 	}
-	return s.upload(bucket, key, buf)
+	return s.upload(bucket, key, buf, withACLBucketOwnerFullControl())
 }
 
 // Upload uploads a file to an S3 bucket under the specified key.
+// The bucket owner, in addition to the object owner, is granted full control.
 func (s *S3) Upload(bucket, key string, data io.Reader) (string, error) {
-	return s.upload(bucket, key, data)
-}
-
-// MkdirTimestamp prefixes the key with the current timestamp "manual/<timestamp>/key".
-func MkdirTimestamp(key string) string {
-	id := time.Now().Unix()
-	return path.Join(artifactDirName, strconv.FormatInt(id, 10), key)
+	return s.upload(bucket, key, data, withACLBucketOwnerFullControl())
 }
 
 // MkdirSHA prefixes the key with the SHA256 hash of the contents of "manual/<hash>/key".
 func MkdirSHA256(key string, content []byte) string {
 	return path.Join(artifactDirName, fmt.Sprintf("%x", sha256.Sum256(content)), key)
 }
 
+// AddonsArtifactPath prefixes the key with the addons artifact path.
+func AddonsArtifactPath(key string) string {
+	return path.Join(artifactDirName, addonDirName, key)
+}
+
+// TemplateArtifactPath prefixes the key with the template artifact path.
+func TemplateArtifactPath(key string) string {
+	return path.Join(artifactDirName, templateDirName, key)
+}
```

Comment:
These don't seem to fit the `s3` package 💭   
What do you think of moving them to `deploy` instead?

Reply:
I moved them to the `template/` package. My motivation was that it might be easier for us to manage/visualize from the codebase what our artifact folder structure would be, if these path definition is put together in a single package.

In the current codebase that are scattered in `cli/deploy`, `deploy` and `s3`.